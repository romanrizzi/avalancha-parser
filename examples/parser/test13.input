
-- Funciones (pre y postcondiciÃ³n)

fun foo1?false

fun foo2!a

fun foo3?a==b!c

fun foo4 : x, y -> z
  ? not (x == y)
  ! f(x) and f(y) imp not f(z)

