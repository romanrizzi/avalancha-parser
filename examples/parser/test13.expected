[
   "program",
   [
      [
         "fun",
         "foo1",
         [
            "sig",
            [],
            "_"
         ],
         [
            "pre",
            [
               "false"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo2",
         [
            "sig",
            [],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "equal",
               [
                  "var",
                  "a"
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ],
         []
      ],
      [
         "fun",
         "foo3",
         [
            "sig",
            [],
            "_"
         ],
         [
            "pre",
            [
               "equal",
               [
                  "var",
                  "a"
               ],
               [
                  "var",
                  "b"
               ]
            ]
         ],
         [
            "post",
            [
               "equal",
               [
                  "var",
                  "c"
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ],
         []
      ],
      [
         "fun",
         "foo4",
         [
            "sig",
            [
               "x",
               "y"
            ],
            "z"
         ],
         [
            "pre",
            [
               "not",
               [
                  "equal",
                  [
                     "var",
                     "x"
                  ],
                  [
                     "var",
                     "y"
                  ]
               ]
            ]
         ],
         [
            "post",
            [
               "imp",
               [
                  "and",
                  [
                     "equal",
                     [
                        "app",
                        "f",
                        [
                           [
                              "var",
                              "x"
                           ]
                        ]
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "app",
                        "f",
                        [
                           [
                              "var",
                              "y"
                           ]
                        ]
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ],
               [
                  "not",
                  [
                     "equal",
                     [
                        "app",
                        "f",
                        [
                           [
                              "var",
                              "z"
                           ]
                        ]
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ]
            ]
         ],
         []
      ]
   ],
   []
]
